src/vos/vos_common.c

```diff
@@ -352,6 +409,16 @@ vos_tx_end(struct vos_container *cont, struct dtx_handle *dth_in,
 			vos_dtx_cleanup_internal(dth);
 	}
 
+	if (dth->dth_local) {
+		for (int i = 0; i < dth->dth_local_oid_cnt; ++i) {
+			vos_cont_decref(dth->dth_local_oid_array[i].dor_cont);
+		}
+
+		dth->dth_local_oid_cnt = 0;
+		D_FREE(dth->dth_local_oid_array);
+		dth->dth_local_oid_cap = 0;
+	}
```

Comment:
Since the array is allocated via vos_dtx_local_begin(), then release such array via vos_dtx_local_end() seems more clean.

Reply:
Done.